### ISSUE TITLE
(IDEA) - What kind of movement should there be in the new HackWars development?

### ISSUE THREAD

**Issue by BWHale1000** [HackWars/docs#3]
I think movement has been a good topic to discuss and definitely something that needs to be discussed and implemented into the new development. I will post comments from the older issue that has since been changed to (IDEAS) so that we have all pertinent comments on this this issue of movement.

Owner (bcoe) suggested the below:
Should we have some fort of world you can walk around in (think rogue or Final Fantasy?).

Owner imR3boot suggested the below:
I don't really know about "walking around" in the game. I do like the idea of the desktop more, although adding a feature as spoken of would be one of a kind and that could be good for the game.

Owner jdog141176 suggested the below:
when you said walking around, i thought of watchdogs right away

i'm gonna try some of the game engines out and see what i can come up with for an rpg style with walking around enabled, maybe we can actually make it work, different houses/places as networks, need a wifi enabled device and or a hotspot.. Coffee shop (starting out place, ofcourse would need to move around to do better hacks maybe? and also a risk of you getting kicked out and or jailed aswell??). Park (would require hotspot or something like..) just some idea's floating around the empty head of mine

Owner jonpchin suggested the below:
If we are going to do walking around then I don't mind a 2D map like in Final Fantasy 2. Or are we talking about something more advanced?

**Comment by BWHale1000** [comment]
I'll add my own comment from the (IDEAS) issue:
You guys talk about "walking around" in this new open source hacking game being developed. I don't think there should be any form of walking around such as a character walking on a map or visiting cities or towns or server rooms or anything like that because of the nature of the game: hacking. You don't move around so much when you are hacking per say, but you are in a single remote location trying to access various things whatever it may be. The only reason moving to places for hacking is involved is if it is proprietary technology that is off the grid or requires you to physically be there in order to access anything. For the "walking around" bit I think one principle should be in mind. Movement. Not just movement on a map as mentioned above, but networking. In the old game all you had to do was click and BOOM, you've switched networks and now can farm those NPCs. That is not how it should be if the idea of changing networks is to be implemented in this new hacking game. I have been working for a WISP company for about two and a half years now and am just getting into running the backbone of the network and everything involved from there on with the current network administrator. There are SO many options for networking it is hard to even think about making people understand it, but I think there are some key elements in networking that can really make HackWars 2.0 a very unique game that is challenging and ever-evolving. There are different layers to networking and there are different protocols that can be used to access different parts of a network. Not only is there that, but you have to have the hardware to handle said protocols, whatever they may be. I think it's a very positive step and key point to the new HackWars. Please ask away as I just did this after work and can easily add some #EDITS or answer any questions.

**Comment by zinapse** [comment]
If anyone here has played the game Street Hacker for PC... I think that type of "movement" would be acceptable. Where you don't actually "move" (W,A,S,D for example) but instead you can click on a map and buy a plane ticket to fly to another place in the world. Just because the player's are "hacking" doesn't mean they shouldn't "move around". When hacking most of the time you are in one area, yes. But there are a lot of reasons to be mobile. Going to a public place for free WiFi so you don't give your personal IP is one of the main things you should do when hacking. But if that were implemented there should obviously be a way around that. Maybe being able to purchase some kind of hardware firewall or IDS that can detect a trace? Hardware being the key though, so it wouldn't be something they have to mess with on the desktop-side.

My two cents. lol

**Comment by bcoe** [comment]
Throw in my two cents too:
- I think the heart of the game should be hacking and programming on a virtual Internet .... but:
- I was thinking it could be really cool to have WIFI as a play mechanic. A quest could have you walk around a city (I'm no artist, so rogue style graphics could be perfect) to find a hidden WIFI network -- perhaps this ultimately leads to finding an underground military facility with a file you need for the over-arching quest.
- well I agree the heart of the game is the hacking and the virtual networks, I think a world could make quests much more powerful.

**Comment by jdog141176** [comment]
i tbh love the idea of having WIFI be a mechanic, would make things more interesting, make it to where you can "ddos" people or something and a virtual world like you stated in a different issue would be amazing!!

**Comment by bcoe** [comment]
Having done a literature review of gaming engines, for our initial requirements, I'm a huge fan of Pixi.js -- It's a really simple API, and I think we could use it to render our game-world, and maybe the inventory/computer screen.

I've put together a demo of the pixi.js engine, so that people can play with it:

https://github.com/HackWars/hackwars-world

**Comment by jonpchin** [comment]
I am just now seeing this topic for the first time. Are we going to be using a click and point map or a keyboard map where users have to move by hitting "WASD" on their keyboard? Or implement both?

**Comment by jdog141176** [comment]
idk since i can't get the demo up there to work at all ( i gotta be doing something wrong :/ ) i can't see what he has made up :(

**Comment by jonpchin** [comment]
Don't worry you are not the only one. I just installed node.js and did the npm install sandcastle from my terminal. I got a bunch of errors. The first error in the stack said "gyp ERR! stack could not find msbuild.exe Do you have Microsoft Visual Studio C++ 2008+ installed?" 

I have no idea why its trying to use my Visual Studios.  I am going to try and get the Node.js plugin for Netbeans and see if I can run the javascripts from there.

**Comment by zinapse** [comment]
(Well I fucked up...I'm using this new Github app, sorry to the person who's comment I deleted...)

It's requiring msbuild.exe because I'm assuming it was compiled using VC. Just install the VC toolchain if you don't want an entire Visual Studio install.

**Comment by jonpchin** [comment]
Yeah I managed to get npm to work. The environment path variable wasn't set so it couldn't find msbuild.exe.

I did some more studying on node.js. Just learned that all those require statements are equivalent to imports. I download Jetbrains to try out as I heard that was the best IDE for javascript. 

To those who are new to node.js the biggest change for me is that node.js programs get run in parallel. So if a slow function has a long run time the node.js runtime environment just continues to run the next line of code instead of getting stuck running one thing command. This means while slow functions are running in the background node.js is processing other requests. Correct me if I am wrong.

**Comment by BWHale1000** [comment]
@jonpchin 
     There aren't any set ideas yet, and I am not going to lie: I just don't see HackWars having rouge-like elements in all honesty. I understand that it is an MMO and there could be some kind of movement but not in the way suggested. I think it would take away from the core of HackWars. Yes, it is a feature that can be added, but what would be the point to it? Moving from wifi connection to wifi connection in different buildings doesnt actually do anything but change your public IP address, and even then, some different buildings have specific settings/equipment to stop any kind of traffic that exceeds their limits or rules. Why not just buy a block of public ip addresses or hack the ISP provider and change your public IP address for a period of time? Besides my rant, can there be 2d/3d movement on a map? Yes. Just not the way it is currently being presented, in my mind of course. I am actually working on a couple different ideas that I haven't posted because they are not thought all the way through yet and may not make total sense if posted half-baked or 3/4 baked.

--EDIT1--
I won't double post so here goes another cent xD I am not against the wifi idea in any way and finding these small range wifi networks for quests is a neat idea. My only wall or worry for this kind of thing is the inevitable map that it will have to function around and how it will correlate wih everything else.
